src/ngraph/runtime/cpu/pass/cpu_mat_fusion.cpp

```diff
@@ -578,16 +578,21 @@ bool runtime::cpu::pass::CPUBatchFusion::run_on_function(std::shared_ptr<Functio
         const Node& node = *n;
         if (TI(node) == TI(op::Concat))
         {
-            auto fused_node = fuse_batch_dot(n);
-            if (fused_node)
+            if (m_fusion_type & DIFFERENTIABLE_FUSIONS)
```

Comment:
seems a bit wasteful to run the whole fuse logic to realize later on it's the wrong fusion type. I'd reverse the checks